### ISSUE TITLE
Environment variables in config not working

### ISSUE THREAD

**Issue by wicol** [Qovery/Replibyte#223]
Should all environment variables work in the config or just the ones you mention in the docs?
`connection_uri: $DATABASE_URL` seems to work but not `connection_uri: postgres://postgres@<my-db-host>:5432/$DB_NAME`:

```sh
$ export DB_NAME=stuff
$ replibyte -c replibyte.yaml dump create
⠤ [00:00:01] [-------------------------------------------------------------------------------------------------------] 0B/100.00MiB (0s)
command error: pg_dump: error: connection to server at "<my-db-host>" (<my-db-ip>), port 5432 failed: FATAL:  database "$DB_NAME" does not exist
```

(`<my...>` is just dedacted)

**Comment by evoxmusic** [comment]
`connection_uri: $DATABASE_URL` will work where interpolation of value with env var will not (e.g `connection_uri: postgres://postgres@<my-db-host>:5432/$DB_NAME`). You can contribute to provide this feature.

**Comment by wicol** [comment]
I don't know rust unfortunately (but I suppose this simple task would be a pretty good start eh? ;) ).

I did however manage to find the code where env var substitution happens and realized it's not that `$DATABASE_URL` is something hardcoded but rather the substitution only supports the entire value being a variable (you might want to point this out in the docs?).
This at least allows me to build the source and destination URLs separately outside of replibyte.

Closing this as it seems current functionality is intended.
